### ISSUE TITLE
Incomplete switch expression with a "case" makes Visual Studio report over ~40+ warnings

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#71957]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Incomplete-switch-expression-with-a-cas/10577466)._

---
[severity:It bothers me. A fix would be nice] [regression] [worked-in:17.4.4]
I'm using Visual Studio 2022 Community 17.8.6.
I was writing a switch expression. I then forgot `case` in a switch expression isn't really a thing in C#, but added it anyway. Instantly, Visual Studio shows like 40 warnings or so above (please note that 40 is just my approximation, it's not an actual count of warnings, but there's a lot of them):
![Error.png](https://aka.ms/dc/image?name=B887941f907ba4ed09eecc864c03d6f0e638425751497275669_Error.png&tid=887941f907ba4ed09eecc864c03d6f0e638425751497275669)

I wouldn't insert a stack trace for every warning, because there's just too many. But here is a stack trace for `CSharpInvertLogicalCodeRefactoringProvider` - in fact, the same exception seems to occur for every warning:
```
System.ArgumentOutOfRangeException : Specified argument was out of the range of valid values.
Parameter name: position
   at Microsoft.CodeAnalysis.SyntaxNode.FindTokenCore(Int32 position,Boolean findInsideTrivia)
   at Microsoft.CodeAnalysis.LanguageService.AbstractHeaderFacts.TryGetAncestorForLocation[TNode](SyntaxNode root,Int32 position)
   at Microsoft.CodeAnalysis.CSharp.LanguageService.CSharpHeaderFacts.IsOnPropertyDeclarationHeader(SyntaxNode root,Int32 position,SyntaxNode& propertyDeclaration)
   at Microsoft.CodeAnalysis.CodeRefactorings.AbstractRefactoringHelpersService`3.<ExtractNodesInHeader>d__11.MoveNext()
   at System.Linq.Enumerable.<OfTypeIterator>d__95`1.MoveNext()
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at Microsoft.CodeAnalysis.CodeRefactorings.AbstractRefactoringHelpersService`3.AddNonHiddenCorrectTypeNodes[TSyntaxNode](IEnumerable`1 nodes,ArrayBuilder`1 resultBuilder,CancellationToken cancellationToken)
   at async Microsoft.CodeAnalysis.CodeRefactorings.AbstractRefactoringHelpersService`3.AddRelevantNodesAsync[TExpressionSyntax,TArgumentSyntax,TExpressionStatementSyntax,TSyntaxNode](<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeRefactorings.AbstractRefactoringHelpersService`3.GetRelevantNodesAsync[TExpressionSyntax,TArgumentSyntax,TExpressionStatementSyntax,TSyntaxNode](<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringContextExtensions.TryGetRelevantNodeAsync[TSyntaxNode](<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.InvertLogical.AbstractInvertLogicalCodeRefactoringProvider`3.ComputeRefactoringsAsync[TSyntaxKind,TExpressionSyntax,TBinaryExpressionSyntax](<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync(<Unknown Parameters>)
```
I'm not exactly sure if this has something to do with Roslyn compilers, but I did receive a warning `Feature 'Document highlights' is currently unavailable due to an internal error` once I typed `case`.

Here is the code that Visual Studio (or Roslyn compiler, I'm not sure) doesn't really like:
```cs
public static float AlphaToFloat(this Argb argb)
        => argb.Alpha.ToString().Length switch
        {
            case
        };
```
In fact, even this causes those warnings:
```cs
public static int X()
    => 5 switch
    {
        case
    }
```
![Error2.png](https://aka.ms/dc/image?name=B061cc02f0286472782072660ab1a4eb0638425755969483175_Error2.png&tid=061cc02f0286472782072660ab1a4eb0638425755969483175)

Removing the incomplete switch expression, saving the file, and restarting Visual Studio removes these warnings. Restarting Visual Studio without removing these
incomplete switch expressions makes the IDE report those warnings again as soon as the editor is rendered.

A fix would be nice.

---
#### Feedback Bot on 2/3/2024, 09:07 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler 

even trying to paste that code in produces:

![image](https://github.com/dotnet/roslyn/assets/4564579/9af10bf8-72aa-4b58-bef1-7217614d0c94)

Seems like some sort of incremental parsing bug at play here.

**Comment by CyrusNajmabadi** [comment]
The problem is here:

![image](https://github.com/dotnet/roslyn/assets/4564579/7bf38b1d-12d3-4945-8545-fda1adb54589)

Specifically:

1. we got an errant 'case'
2. we attached it to an empty pattern as skipped leading trivia)
3. we tried to parse out a switch arm
4. we think teh arm's length is 0 (despite teh skipped syntax)
5. we break out.

so we've consumed a 'case' that isn't present in the tree.

teh likely fix here is that we need to not look at .Width, but instead look at .FullWidth.  That way if there is *anything* even skipped syntax, we consume it.  Otherwise, if we want thsi logic, we'll need to do it in a reset block and rewind if we take nothing.
